tests/integration/results.go

```diff
@@ -68,6 +80,55 @@ func areResultsAnyOf(expected AnyOf, actual any) bool {
 	return false
 }
 
+// UniqueCid allows the referencing of Cids by an arbitrary test-defined ID.
+//
+// Instead of asserting on a specific Cid value, this type will assert that
+// no other [UniqueCid]s with different [ID]s has the first Cid value that this instance
+// describes.
+//
+// It will also ensure that all Cids described by this [UniqueCid] have the same
+// valid, Cid value.
+type UniqueCid struct {
+	// ID is the arbitrary, but hopefully descriptive, id of this [UniqueCid].
+	ID any
```

Comment:
suggestion: I hate `any`. I believe this be the last resort in all cases as it kills strong typing.
In your examples you use `string`, so why not to make it `string` or `int`?